### PR TITLE
Get hotspots from server in SQ Client

### LINKED ISSUE #4223: Get hotspots from server in SQ Client

### PR DESCRIPTION
**Pull request by ugras-ergun-sonarsource**
fixes #4223

### CODE REVIEW

**Comment by ugras-ergun-sonarsource** [comment]
QG failed due to missing tests, but there's no business logic in CTOR and adding tests to just satisfy QG seems useless.

**Comment by georgii-borovinskikh-sonarsource** [comment]
No deserialization in this PR?

**Comment by georgii-borovinskikh-sonarsource** [comment]
IMO you should add deserialization. Will also improve code coverage

**Review by georgii-borovinskikh-sonarsource — CHANGES_REQUESTED** [review]
I think it would be better to add json deserialization and tests to this PR

**Comment by ugras-ergun-sonarsource** [comment]
There's a seperate ticket for that. These first tickets are for creating the public facing members first and removing the dependency. So we can use it in other tasks.

**Comment by georgii-borovinskikh-sonarsource** [comment]
We are not really blocked by it now, it's going to be needed way later + it's hard for me to actually compare these models to the server response jsons without tests

File: src/SonarQube.Client/Api/Common/ServerComponent.cs
Comment (ugras-ergun-sonarsource) [inline src/SonarQube.Client/Api/Common/ServerComponent.cs:1]: I moved this file out of getissuesrequest. So we will not create duplication or dependenct between requests.

File: src/SonarQube.Client/Api/Common/ServerIssueTextRange.cs
Comment (ugras-ergun-sonarsource) [inline src/SonarQube.Client/Api/Common/ServerIssueTextRange.cs:1]: I moved this file out of getissuesrequest. So we will not create duplication or dependenct between requests.

File: src/SonarQube.Client.Tests/Requests/Api/V9_7/SearchHotspotRequestTests.cs [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+using SonarQube.Client.Tests.Infra;
+using static SonarQube.Client.Tests.Infra.MocksHelper;
+
+namespace SonarQube.Client.Tests.Requests.Api.V9_7
+{
+    [TestClass]
+    public class SearchHotspotRequestTests
+    {
+        [TestMethod]
+        public async Task InvokeAsync_NormalizesPath()
```
Comment (ugras-ergun-sonarsource) [inline src/SonarQube.Client.Tests/Requests/Api/V9_7/SearchHotspotRequestTests.cs:38]: As the convention I will create more detailed tests for checking all fields in the next task (creating an endpoint)

File: src/SonarQube.Client/Api/ISearcHotspotRequest.cs [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using SonarQube.Client.Models;
+using SonarQube.Client.Requests;
+
+namespace SonarQube.Client.Api
+{
+    public interface ISearcHotspotRequest : IRequest<SonarQubeHotspotSearch[]>
```
Comment (georgii-borovinskikh-sonarsource) [inline src/SonarQube.Client/Api/ISearcHotspotRequest.cs:26]: typo

File: src/SonarQube.Client/Api/Common/ServerComponent.cs [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+        [JsonProperty("path")]
+        public string Path { get; set; }
+
+        public bool IsFile
+        {
+            get { return Qualifier == "FIL"; }
+        }
+
+        internal static ILookup<string, string> GetComponentKeyPathLookup(JObject root)
```
Comment (georgii-borovinskikh-sonarsource) [inline src/SonarQube.Client/Api/Common/ServerComponent.cs:44]: Maybe a separate class for this helper?

**Review by georgii-borovinskikh-sonarsource — APPROVED** [review]
A couple of small comments, otherwise LGTM

File: src/SonarQube.Client/Api/Common/ServerComponent.cs (hunk 3 above)
Comment (ugras-ergun-sonarsource) [reply src/SonarQube.Client/Api/Common/ServerComponent.cs]: Good idea I have changed them to extension methods to increase readability.

**Pull request merged**